retrieve/package.go

```diff
@@ -0,0 +1,10 @@
+// Package retrieve assists buildpack authors to create the retrieve step as outlined in the [Dependency RFCs],
+// especially `Dependency Management Phase 2: Workflow and Github Action Generalization RFC`.
+//
+// Buildpacks should call retrieve.NewMetadata from the `main` func of their retrieval step and pass in the necessary
+// function implementations:
+// - libdependency.VersionFetcherFunc: to retrieve all versions of a dependency from the internet
+// - retrieve.GenerateMetadataFunc: to generate metadata for a specific version
+//
+// [Dependency RFCs]: https://github.com/paketo-buildpacks/rfcs/tree/main/text/dependencies/rfcs
+package retrieve
```

Comment:
I like this description, thanks for adding it. Is there a reason it goes in a file like this rather than just a README in the `retrieve` directory?

Reply:
Hmmmm.... The whole library does struggle a little bit with documentation and where it's located. I prefer docs to be located as close as possible to the code, and `godoc` feels a little closer than a `README` even if it's a little less discoverable. Plus `godoc` has the added perk that Goland will automatically refactor symbol names embedded in the `godoc`.